src/agent/src/main.rs

```diff
@@ -199,6 +204,17 @@ async fn real_main() -> std::result::Result<(), Box<dyn std::error::Error>> {
         ttrpc_log_guard = Ok(slog_stdlog::init().map_err(|e| e)?);
     }
 
+    if config.tracing != tracer::TraceType::Disabled {
+        let _ = tracer::setup_tracing(NAME, &logger, &config)?;
```

Comment:
I noticed `instrument` is used with functions called prior to setting up tracing. Is this intentional?

Reply:
Thanks - this should be benign, but is unnecessary so I've removed 'em ;)